src/Psl/DateTime/DateTimeConvenienceMethodsTrait.php

```diff
@@ -346,7 +347,7 @@ public function getISOWeekNumber(): array
      */
     public function getWeekday(): Weekday
     {
-        return Weekday::from((int) $this->format(pattern: 'e'));
+        return Weekday::from((int) $this->format(pattern: 'e', locale: Locale::EnglishUnitedKingdom));
```

Comment:
Turns out these expected formats might give different results for different locales.
Hardcoded the en_GB way as it was used in the tests and is considered correct for these cases.